ambry-api/src/main/java/com.github.ambry/config/ConnectionPoolConfig.java

```diff
@@ -60,6 +60,14 @@
   @Default("2")
   public final int connectionPoolMaxConnectionsPerPortSSL;
 
+  /**
+   * The maximum consecutive connect failures allowed before we clean up the available pool
+   */
+  @Config("connectionpool.max.connect.failures.to.clean.up.available.pool")
```

Comment:
I am suggesting getting rid of this config altogether. If we need a threshold based mechanism, we should go for approach 2.